master/internal/rm/kubernetesrm/job.go

```diff
@@ -143,284 +140,363 @@ func newPod(
 		masterTLSConfig:      masterTLSConfig,
 		loggingTLSConfig:     loggingTLSConfig,
 		loggingConfig:        loggingConfig,
-		slots:                msg.Slots,
+		numPods:              msg.NumPods,
+		slotsPerPod:          msg.Slots,
 		podInterface:         podInterface,
 		configMapInterface:   configMapInterface,
 		resourceRequestQueue: resourceRequestQueue,
-		podName:              uniqueName,
+		jobName:              uniqueName,
 		configMapName:        uniqueName,
-		container:            podContainer,
+		podNodeNames:         make(map[string]string),
+		container: cproto.Container{
+			ID:          cproto.ID(msg.Spec.ContainerID),
+			State:       cproto.Assigned,
+			Description: msg.Spec.Description,
+		},
 		containerNames:       containerNames,
 		scheduler:            scheduler,
 		slotType:             slotType,
 		slotResourceRequests: slotResourceRequests,
-		syslog: logrus.New().WithField("component", "pod").WithFields(
+		syslog: logrus.New().WithField("component", "job").WithFields(
 			logger.MergeContexts(msg.LogContext, logger.Context{
-				"pod": uniqueName,
+				"job": uniqueName,
 			}).Fields(),
 		),
 	}
 	return p
 }
 
-func (p *pod) start() error {
-	if p.restore {
-		if p.container.State == cproto.Running {
-			err := p.startPodLogStreamer()
-			if err != nil {
+func (j *job) start() error {
+	if j.restore {
+		if j.container.State == cproto.Running {
+			if err := j.startPodLogStreamers(); err != nil {
 				return err
 			}
 		}
 	} else {
-		if err := p.createPodSpecAndSubmit(); err != nil {
+		if err := j.createSpecAndSubmit(); err != nil {
 			return fmt.Errorf("creating pod spec: %w", err)
 		}
 	}
 	return nil
 }
 
-func (p *pod) finalize() {
-	p.kill()
-	p.finalizeTaskState()
+func (j *job) finalize() {
+	j.mu.Lock()
+	defer j.mu.Unlock()
+
+	// If an error occurred during the lifecycle of the pods, we need to update the scheduler
+	// and the task handler with new state.
+	if j.container.State != cproto.Terminated {
+		j.kill()
+		j.syslog.Warnf("killed job after our handler exited unexpectedly")
```

Comment:
nit: "killed job after pod handler exited unexpectedly"

Reply:
i dont understand. what it says it what i mean for it to say. there is no "pod handler" anymore.